### PR TITLE
Refactor the traitor panel to be based off of antag datums

### PR DESCRIPTION
**Pull request by ghost**
- [x] Traitor
- [x] Blood Brothers
- [x] Cult
- [x] Clock Cult
- [x] Abductors
- [x] Monkey
- [x] Ninja
- [x] Wizard
- [x] Nuke Ops
- [x] Devil
- [x] Changeling

- [x] Actually make the traitor panel hrefs work

### CODE REVIEW

**Comment by praisenarsie** [comment]
@AnturK

**Comment by MoreRobustThanYou** [comment]
It works now

**Comment by MoreRobustThanYou** [comment]
@Cyberboss why does need pictures? the traitor panel looks the same

**Comment by Cyberboss** [comment]
It also shows you tested it

**Comment by MoreRobustThanYou** [comment]
ok

**Comment by MoreRobustThanYou** [comment]
@Cyberboss 

capture
capture2

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -13,12 +13,17 @@ SUBSYSTEM_DEF(job)
 	var/list/prioritized_jobs = list()
 	var/list/latejoin_trackers = list()	//Don't read this list, use GetLateJoinTurfs() instead
 
+	var/list/antag_datum_cache = list()
```
Comment (AnturK) [inline code/controllers/subsystem/job.dm]: This has nothing to do with jobs.

**Review by AnturK — CHANGES_REQUESTED** [review]
The panel needs more than just low-hanging fruit changes only

File: code/datums/antagonists/changeling.dm [hunk 2]
```diff
@@ -75,6 +80,10 @@
 			forge_team_objectives()
 		forge_objectives()
 	remove_clownmut()
+	if(owner && !stuff_generated) //avoid runtiming when creating an ownerless datum
```
[the remark this answers is not in the document]
Comment (MoreRobustThanYou) [reply code/datums/antagonists/changeling.dm]: woops!

File: code/controllers/subsystem/job.dm [hunk 3]
```diff
@@ -13,12 +13,17 @@ SUBSYSTEM_DEF(job)
...
+
 /datum/controller/subsystem/job/Initialize(timeofday)
 	if(!occupations.len)
 		SetupOccupations()
 	if(CONFIG_GET(flag/load_jobs_from_txt))
 		LoadJobs()
 	generate_selectable_species()
+	for(var/_A in subtypesof(/datum/antagonist))
+		var/datum/antagonist/A = new _A()
```
[the remark this answers is not in the document]
Comment (MoreRobustThanYou) [reply code/controllers/subsystem/job.dm]: I mean, the only datum that actually caused problems when made without an owner was the changeling datum, and i fixed that

File: code/datums/antagonists/abductor.dm [hunk 4]
```diff
@@ -105,6 +106,41 @@
...
 	return result.Join("<br>")
 
+/datum/antagonist/abductor/antag_panel_section(datum/mind/mind, mob/current)
+	if(!ishuman(current))
+		return FALSE
+	var/text = "abductor"
+	if(SSticker.mode.config_tag == "abductor")
+		text = uppertext(text)
+	text = "<i><b>[text]</b></i>: "
+	if(src in get_antagonists(/datum/antagonist/abductor))
```
[the remark this answers is not in the document]
Comment (MoreRobustThanYou) [reply code/datums/antagonists/abductor.dm]: woops!

**Pull request closed without merging**